src/telemetry/Manager.cc

```diff
@@ -38,6 +43,232 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
 	pimpl.swap(ptr);
 	}
 
+// -- collect metric stuff -----------------------------------------------------
+
+const std::vector<std::string_view> Manager::Metric::label_names() const
+	{
+	std::vector<std::string_view> r;
+
+	// Labels are backed by strings, so string_view should be safe.
+	for ( auto const& l : broker::telemetry::label_names(family_) )
+		r.push_back(l);
+
+	return r;
+	}
+
+Manager::Metric::Metric(Manager::Metric::MetricType metric_type,
+                        const broker::telemetry::metric_family_hdl* family,
+                        std::optional<double> value, std::optional<int64_t> count_value,
+                        std::vector<std::string_view> labels)
+	: metric_type_(metric_type), family_(family), labels_(labels), value_(value),
+	  count_value_(count_value)
+	{
+	}
+
+zeek::RecordValPtr Manager::Metric::get_metric_opts_record()
+	{
+	static auto svt = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto rt = zeek::id::find_type<zeek::RecordType>("Telemetry::MetricOpts");
+	auto r = make_intrusive<zeek::RecordVal>(rt);
+
+	static auto rt_off_prefix = rt->FieldOffset("prefix");
+	static auto rt_off_name = rt->FieldOffset("name");
+	static auto rt_off_helptext = rt->FieldOffset("helptext");
+	static auto rt_off_unit = rt->FieldOffset("unit");
+	static auto rt_off_is_total = rt->FieldOffset("is_total");
+	static auto rt_off_labels = rt->FieldOffset("labels");
+	static auto rt_off_bounds = rt->FieldOffset("bounds");
+	static auto rt_off_metric_type = rt->FieldOffset("metric_type");
+
+	r->Assign(rt_off_prefix, make_intrusive<zeek::StringVal>(prefix()));
+	r->Assign(rt_off_name, make_intrusive<zeek::StringVal>(name()));
+	r->Assign(rt_off_helptext, make_intrusive<zeek::StringVal>(helptext()));
+	r->Assign(rt_off_unit, make_intrusive<zeek::StringVal>(unit()));
+	r->Assign(rt_off_is_total, val_mgr->Bool(is_sum()));
+
+	auto label_names_vec = make_intrusive<zeek::VectorVal>(svt);
+	int i = 0;
+	for ( auto const& l : label_names() )
+		label_names_vec->Assign(i++, make_intrusive<StringVal>(l));
+	r->Assign(rt_off_labels, label_names_vec);
+
+	auto et = zeek::id::find_type<zeek::EnumType>("Telemetry::MetricType");
+	zeek_int_t metric_type_int = -1;
+
+	// This is mapping internal enum values for the types to the
+	// right EnumVal. TODO: There might be a better way.
+	switch ( metric_type_ )
```

Comment:
I don't think you can get around the switch, but you can declare an enum type in a bif and then use those constants instead of the `Lookups`, then this should compress quite a bit.  See, e.g., `BrokerProtocol` in `src/broker/comm.bif`

Reply:
I attempted this. While it didn't compress the Lookup calls are gone. But I may have missed something.